src/components/column/Column.tsx

```diff
@@ -0,0 +1,208 @@
+import { ColumnObjectType, ItemObjectType } from "../../types";
+import { TextAreaWrapper } from "../card/card.styles";
+
+type BoardColumnProps = {
+  key: string;
+  column: ColumnObjectType;
+  items: ItemObjectType[];
+  index: number;
+};
+
+export const Column: React.FC<BoardColumnProps> = (props) => {
+  const { dispatch } = useBoard();
+  const [addField, setAddField] = useState<boolean>(false);
+  const [titleToggle, setTitleToggle] = useState<boolean>(false);
+  const [addCardField, setAddCardField] = useState<boolean>(false);
+  const [title, setTitle] = useState<string>("");
+  const [columnTitle, setColumnTitle] = useState<string>(props.column.title);
+  const [content, setContent] = useState<string>("");
+  const [description, setDescription] = useState<string>("");
+
+  const addClick = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+    setAddField(true);
+  },[addField]);
+
+  const addCardClick = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+    setAddCardField(true);
+  }, [addCardField]);
+
+  const addItem = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if(Boolean(description) && Boolean(content)) {
+      dispatch({
+        type: "ADD_ITEM",
+        payload: { column: props.column.id, item: { content, id: nanoid(), description, updated: new Date(Date.now()) } },
+      });
+      setAddField(false);
+    }
+
+  };
+  const editColumnTitle = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if(Boolean(columnTitle)){
+      dispatch({
+        type: "EDIT_COLUMN_TITLE",
+        payload: { column: props.column.id, title: { columnTitle } },
+      });
+      setTitleToggle(false);
+    }
+
+  };
+
+  const addCard = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if(Boolean(title)){
+      dispatch({
+        type: "ADD_CARD",
+        payload: { column: { title: title, id: nanoid(), itemsIds: [] } },
+      });
+      setAddCardField(false);
+    }
+
+  };
+
+  return (
+
+    <Draggable draggableId={props.column.id} index={props.index}>
+      {(provided: DraggableProvided, snapshot: DraggableStateSnapshot) => (
+
+          <BoardColumnWrapper
+              {...provided.draggableProps}
+              ref={provided.innerRef}
+          >
+            {props.column.title ? (
+                <React.Fragment>
+                  {!titleToggle ? (
+                      <TitleWrapper>
+                        <BoardColumnTitle
+                            {...provided.dragHandleProps}>{props.column.title}</BoardColumnTitle>
+                        <span>
+                <ButtonWIthIcon
+                    icon="fa fa-pencil"
+                    callback={(e) => {
+                      setTitleToggle(true);
+                    }}
+                />
+              </span>
+                      </TitleWrapper>
+                  ) : (
+                      <TextAreaWrapper>
+                        <i
+                            onClick={(e) => {
```

Comment:
Why this function is inline why we dont usecall back

Reply:
Changed at every instance in the app.